application/controllers/Station.php

```diff
@@ -208,4 +208,17 @@ public function get_county() {
         echo json_encode($json);
     }
 
+    // Function return db value from station_profil field //
+	public function get_json_info() {
+		$return_json = array();
+		$this->load->model('stations');
+		$station_id = $this->input->post('station_id'); 
+		$field = $this->input->post('field');
+		if (($station_id>0) && (!empty($field))) {
+			$return_json[$field] = $this->stations->get_station_info($station_id,$field,"");
+		}
```

Comment:
See comment at Qso.php.
- get_station_info is not nescessary. it's redundant to profile.
- Please CHECK that station_id belongs to user
- dynamic-fields ($field) is not a good idea. If User/Attacker puts in "qrzapikey" as field, it'll be exposed. Security-Risk!

Reply:
field is not dynamic, it was checked on exist field in table  `$fields_allow = array('eqsl_defaultqslmsg');`
with this method, we can avoid using several functions for the same final action (return 1 value).
But i wil change to profile(), so

Reply:
yes, that was the idea behind "profile". it fetches all at the model-layer. and at controller you can specify which one is needed and if the key is granted and so on.
the power-thing is a historic relict ;)